subsys/spm/secure_services.c

```diff
@@ -143,3 +144,16 @@ int spm_firmware_info(u32_t fw_address, struct fw_info *info)
 	return -EFAULT;
 }
 #endif /* CONFIG_SPM_SERVICE_FIND_FIRMWARE_INFO */
+
+
+#ifdef CONFIG_SPM_SERVICE_PREVALIDATE
+__TZ_NONSECURE_ENTRY_FUNC
+int spm_prevalidate_b1_upgrade(u32_t dst_addr, u32_t src_addr)
+{
+	if (!bl_validate_firmware_available()) {
```

Comment:
Could we check the version numbers in the slots and deduce what the dst_addr is from that? Or is it more flexible to let the user caller handle this?

Reply:
If the user has an upgrade, they could even call it once with dst_addr as S0, and if that fails, call it with S1.